Python/Product/VSInterpreters/StringListReader.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.PythonTools.Interpreter {
+    class StringListReader : TextReader {
+        private readonly IEnumerator<string> _strings;
+        private StringReader _current;
+        private int _peekBuffer = -1;
+
+        public StringListReader(IEnumerable<string> strings) {
+            _strings = strings.GetEnumerator();
+            if (_strings.MoveNext()) {
+                _current = new StringReader(_strings.Current);
+            }
+        }
+
+        protected override void Dispose(bool disposing) {
+            base.Dispose(disposing);
+            if (disposing) {
+                _strings.Dispose();
+            }
+        }
+
+        private bool Next() {
+            if (_current != null) {
+                if (_strings.MoveNext()) {
+                    _current = new StringReader(_strings.Current);
+                    return true;
+                } else {
+                    _current = null;
+                }
+            }
+            return false;
+        }
+
+        public override int Peek() {
+            if (_peekBuffer >= 0) {
+                return _peekBuffer;
+            }
+            _peekBuffer = Read();
+            return _peekBuffer;
+        }
+
+        public override string ReadLine() {
+            if (_peekBuffer == 0) {
```

Comment:
What's the special significance of 0? I can't see any place that sets `_peekBuffer` to that value.

Reply:
If we read a null byte. Things are going to get weird if we prefix a string with `\0`

Reply:
Actually, now I think about it, that doesn't matter. I spent too long in C last week...